### DISCUSSION TITLE
Ruffle player on own html page?

### DISCUSSION THREAD

**Discussion by Varaosa** [ruffle-rs/ruffle#3399] [General]
Hello,

I like to ask how i could add ruffle player into own web page to play flash file using ruffle engine?

Are there any source with flash example how to play flash on normal html file, that have flash content?

What i actually want then, i like to use local html file to play flash content in web browser.

**Comment by ousia** [comment]
@Varaosa, a minimal sample can be found at https://ousia.github.com/ruffle-demo/

Download the selfhosted version (the file ending with `-selfhosted.zip`) from https://github.com/ruffle-rs/ruffle/releases.

The relevant HTML code reads:

``` html
<object data="./006699.swf" type="application/x-shockwave-flash" width="400" height="400">
  <param name="movie" value="./006a699.swf" />
  <param name="allowScriptAccess" value="sameDomain" />
  <param name="quality" value="high" />
  <param name="scale" value="noscale" />
  <param name="menu" value="false" />
</object>

<script src="./ruffle.js"></script>
```

Repository @ https://github.com/ousia/ruffle-demo.

I hope this answers your question.

**Reply by Varaosa** [reply]
> 
> 
> @Varaosa, a minimal sample can be found at https://ousia.github.com/ruffle-demo/
> 
> Download the selfhosted version (the file ending with `-selfhosted.zip`) from https://github.com/ruffle-rs/ruffle/releases.
> 
> The relevant HTML code reads:
> 
> ```
> <object data="./006699.swf" type="application/x-shockwave-flash" width="400" height="400">
>   <param name="movie" value="./006a699.swf" />
>   <param name="allowScriptAccess" value="sameDomain" />
>   <param name="quality" value="high" />
>   <param name="scale" value="noscale" />
>   <param name="menu" value="false" />
> </object>
> 
> <script src="./ruffle.js"></script>
> ```
> 
> Repository @ https://github.com/ousia/ruffle-demo.
> 
> I hope this answers your question.

-Try this but, i got error:

Someting went wrong :c

It appears you running Ruffle as the "file:" protocol.
This doesn't work as browsers block many fatures from working for security reasons.
Instaead, we invite you setup a local server or either use the web demo or desktop application.

So, i think i din't doing this right or someting but, it's not open or work my Firefox browser with those script. What i need do more or someting else to play my file local in web browser using Ruffle player?

**Reply by ousia** [reply]
Could you unzip [minimal-ruffle.zip](https://github.com/ruffle-rs/ruffle/files/6036844/minimal-ruffle.zip) and open `index.html`?

This works for me perfectly fine in _Firefox_.

**Reply by danielhjacobs** [reply]
On Firefox, you may need to disable same origin security: Go to about:config, accept the risks, search security.fileuri.strict_origin_policy, and change it to false. This is dangerous for other webpages, so you may want to switch it back to true whenever you're not playing a Flash file. On Chrome, there is no setting to disable this security feature, so you need to set up a minimal web server. You can add to Chrome the "Web Server for Chrome" extension (https://chrome.google.com/webstore/detail/web-server-for-chrome/ofhbbkphhbklhfoeikjpcbhemlocgigb), click Launch App once it's installed, click "Choose Folder", navigate to the folder on your PC with your HTML and SWF file, and with it highlighted click "Select Folder". Then your files will be accessible from the web server URL (running on a specified port on 127.0.0.1 - a server IP better known as localhost) listed on the Web Server extension.

**Reply by Varaosa** [reply]
> 
> 
> Could you unzip [minimal-ruffle.zip](https://github.com/ruffle-rs/ruffle/files/6036844/minimal-ruffle.zip) and open `index.html`?
> 
> This works for me perfectly fine in _Firefox_.



> 
> 
> On Firefox, you may need to disable same origin security: Go to about:config, accept the risks, search security.fileuri.strict_origin_policy, and change it to false. This is dangerous for other webpages, so you may want to switch it back to true whenever you're not playing a Flash file. On Chrome, there is no setting to disable this security feature, so you need to set up a minimal web server. You can add to Chrome the "Web Server for Chrome" extension (https://chrome.google.com/webstore/detail/web-server-for-chrome/ofhbbkphhbklhfoeikjpcbhemlocgigb), click Launch App once it's installed, click "Choose Folder", navigate to the folder on your PC with your HTML and SWF file, and with it highlighted click "Select Folder". Then your files will be accessible from the web server URL (running on a specified port on 127.0.0.1 - a server IP better known as localhost) listed on the Web Server extension.

-Yes, that fix my issue for FireFox & thank for the info! :3

But, i got another issue that was about Chrome browser or pluging. I don't have or use Goolge Chrome on my pc but, i have software that use Goolge Chrome plugin. 

I use software named "Wallepaper Engine" -on Steam platform & i got project there that use html file there & use flash content there but, issue is that then this Wallepaper Engine use Google Chrome pluging to play my flash file & i can't play my current works there.

Project work in FireFox now but, i want to get this work also in "Wallepaper Engine". This show this ruffle error that i got there right now but, just because "Wallepaper Engine" -Google Chrome pluging not have support Flash or Ruffle Player.

How i solute this issue? It's issue in "Wallepaper Engine" or do i need to do someting else setup or change to allow Google Chrome pluging use Flash? 

How i do it because, i don't have Google Chrome on pc installed. I have only Chrome pluging on pc but, how i allow Chrome to false security.fileuri.strict_origin_policy if i don't have Chrome on my pc?

**Comment by Varaosa** [comment]
What Wallepaper Engine support say:
Please ask the Ruffle developers how to configure Chromium....

So, how to configure Chromium on Ruffle or use Ruffle if you don't have Chromium installed on pc?

In this case if you want to show HTML -file with flash content in Chromium, even if you don't have Chromium installed on pc.
